File: Level 1/택배 상자 꺼내기/solution.py
def solution(n, w, num):
    height = n // w
    box_map = []

    for i in range(height + 1):
        row = [j + (w * i) for j in range(1, w + 1)]
        box_map.append(row if i % 2 == 0 else row[::-1])

    floor = num // w - 1 if num % w == 0 else num // w
    row_index = box_map[floor].index(num)

    answer = 1

    while floor < height:
        floor += 1
        if box_map[floor][row_index] <= n:
            answer += 1

    return answer

File: Level 1/택배 상자 꺼내기/solution.js
function solution(n, w, num) {
  let isHaveToPopLastBox = true;
  const height = Math.ceil(n / w);
  const targetHeight = Math.ceil(num / w);
  const nRemain = n % w || n;
  const numRemain = num % w || num;

  if (height % 2 === targetHeight % 2 && nRemain < numRemain)
    isHaveToPopLastBox = false;
  if (height % 2 !== targetHeight % 2 && nRemain + numRemain <= w)
    isHaveToPopLastBox = false;

  return height - targetHeight + Number(isHaveToPopLastBox);
}
